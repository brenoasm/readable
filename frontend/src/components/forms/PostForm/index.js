import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Input from '../Input';
import Select from '../Select';
import TextArea from '../TextArea';

const propTypes = {
  properties: PropTypes.object,
  handleInput: PropTypes.func
};

const defaultProps = {
  properties: {},
  handleInput: () => {}
};

const StyledPostForm = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
`;

const PostForm = ({ properties, handleInput }) => {
  return (
    <StyledPostForm>
      <Input
        title="Autor: "
        name={properties.author.name}
        value={properties.author.value}
        errors={properties.author.errors}
        handleChange={handleInput}
      />
      <Input
        title="Titulo: "
        name={properties.title.name}
        value={properties.title.value}
        errors={properties.title.errors}
        handleChange={handleInput}
      />
      <Select
        title="Categoria: "
        name={properties.category.name}
        value={properties.category.value}
        errors={properties.category.errors}
        handleChange={handleInput}
        options={properties.category.options}
      />
      <TextArea
        title="Descrição: "
        name={properties.body.name}
        value={properties.body.value}
        errors={properties.body.errors}
        handleChange={handleInput}
      />
    </StyledPostForm>
  )
};

PostForm.propTypes = propTypes;
PostForm.defaultProps = defaultProps;

export default PostForm;
