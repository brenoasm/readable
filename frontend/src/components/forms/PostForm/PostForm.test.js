/* eslint-disable */

import React from 'react';

import PostForm from './';

describe('PostForm', () => {
  const initialState = {
    properties: {
      id: {
        isFormField: false,
        value: '',
        originalValue: '',
        isDirty: false,
        isValid: false,
        errors: [],
        name: 'id'
      },
      timestamp: {
        isFormField: false,
        value: '',
        originalValue: '',
        isDirty: false,
        isValid: false,
        errors: [],
        name: 'timestamp'
      },
      title: {
        validations: [],
        isFormField: true,
        isValid: false,
        value: '',
        originalValue: '',
        isDirty: false,
        errors: [],
        name: 'title'
      },
      body: {
        validations: [],
        isFormField: true,
        isValid: false,
        value: '',
        originalValue: '',
        isDirty: false,
        errors: [],
        name: 'body'
      },
      author: {
        validations: [],
        isFormField: true,
        isValid: false,
        value: '',
        originalValue: '',
        isDirty: false,
        errors: [],
        name: 'author'
      },
      category: {
        validations: [],
        isFormField: true,
        isValid: false,
        options: [
          {
            label: 'react',
            value: 'react'
          },
          {
            label: 'redux',
            value: 'redux'
          },
          {
            label: 'udacity',
            value: 'udacity'
          }
        ],
        value: '',
        originalValue: '',
        isDirty: false,
        errors: [],
        name: 'category'
      }
    }
  };

  const mountedWrapper = mount(<PostForm {...initialState} />);

  const names = mountedWrapper
      .find('input')
      .reduce((total, value) => [...total, value.prop('name')], []);

  it('matches the snapshot', () => {
    const wrapper = shallow(<PostForm {...initialState} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should a input of name author', () => {
    expect(names).toContain('author');
  });

  it('should a input of name title', () => {
    expect(names).toContain('title');
  });

  it('should a select of name category', () => {
    expect(mountedWrapper.find('select').prop('name')).toBe('category');
  });

  it('should a textarea of name body', () => {
    expect(mountedWrapper.find('textarea').prop('name')).toBe('body');
  });
});
