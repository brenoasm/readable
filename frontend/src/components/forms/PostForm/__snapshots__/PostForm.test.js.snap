// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostForm
    handleInput={[Function]}
    properties={
        Object {
            "author": Object {
              "errors": Array [],
              "isDirty": false,
              "isFormField": true,
              "isValid": false,
              "name": "author",
              "originalValue": "",
              "validations": Array [],
              "value": "",
            },
            "body": Object {
              "errors": Array [],
              "isDirty": false,
              "isFormField": true,
              "isValid": false,
              "name": "body",
              "originalValue": "",
              "validations": Array [],
              "value": "",
            },
            "category": Object {
              "errors": Array [],
              "isDirty": false,
              "isFormField": true,
              "isValid": false,
              "name": "category",
              "options": Array [
                Object {
                  "label": "react",
                  "value": "react",
                },
                Object {
                  "label": "redux",
                  "value": "redux",
                },
                Object {
                  "label": "udacity",
                  "value": "udacity",
                },
              ],
              "originalValue": "",
              "validations": Array [],
              "value": "",
            },
            "id": Object {
              "errors": Array [],
              "isDirty": false,
              "isFormField": false,
              "isValid": false,
              "name": "id",
              "originalValue": "",
              "value": "",
            },
            "timestamp": Object {
              "errors": Array [],
              "isDirty": false,
              "isFormField": false,
              "isValid": false,
              "name": "timestamp",
              "originalValue": "",
              "value": "",
            },
            "title": Object {
              "errors": Array [],
              "isDirty": false,
              "isFormField": true,
              "isValid": false,
              "name": "title",
              "originalValue": "",
              "validations": Array [],
              "value": "",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Input
          className=""
          errors={Array []}
          handleChange={[Function]}
          id=""
          name="author"
          placeholder=""
          title="Autor: "
          type="text"
          value=""
/>,
        <Input
          className=""
          errors={Array []}
          handleChange={[Function]}
          id=""
          name="title"
          placeholder=""
          title="Titulo: "
          type="text"
          value=""
/>,
        <Select
          className=""
          errors={Array []}
          handleChange={[Function]}
          id=""
          name="category"
          options={
                    Array [
                              Object {
                                "label": "react",
                                "value": "react",
                              },
                              Object {
                                "label": "redux",
                                "value": "redux",
                              },
                              Object {
                                "label": "udacity",
                                "value": "udacity",
                              },
                            ]
          }
          title="Categoria: "
          value=""
/>,
        <TextArea
          className=""
          errors={Array []}
          handleChange={[Function]}
          id=""
          name="body"
          title="Descrição: "
          value=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "",
          "errors": Array [],
          "handleChange": [Function],
          "id": "",
          "name": "author",
          "placeholder": "",
          "title": "Autor: ",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "",
          "errors": Array [],
          "handleChange": [Function],
          "id": "",
          "name": "title",
          "placeholder": "",
          "title": "Titulo: ",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "",
          "errors": Array [],
          "handleChange": [Function],
          "id": "",
          "name": "category",
          "options": Array [
            Object {
              "label": "react",
              "value": "react",
            },
            Object {
              "label": "redux",
              "value": "redux",
            },
            Object {
              "label": "udacity",
              "value": "udacity",
            },
          ],
          "title": "Categoria: ",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "",
          "errors": Array [],
          "handleChange": [Function],
          "id": "",
          "name": "body",
          "title": "Descrição: ",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Input
            className=""
            errors={Array []}
            handleChange={[Function]}
            id=""
            name="author"
            placeholder=""
            title="Autor: "
            type="text"
            value=""
/>,
          <Input
            className=""
            errors={Array []}
            handleChange={[Function]}
            id=""
            name="title"
            placeholder=""
            title="Titulo: "
            type="text"
            value=""
/>,
          <Select
            className=""
            errors={Array []}
            handleChange={[Function]}
            id=""
            name="category"
            options={
                        Array [
                                    Object {
                                      "label": "react",
                                      "value": "react",
                                    },
                                    Object {
                                      "label": "redux",
                                      "value": "redux",
                                    },
                                    Object {
                                      "label": "udacity",
                                      "value": "udacity",
                                    },
                                  ]
            }
            title="Categoria: "
            value=""
/>,
          <TextArea
            className=""
            errors={Array []}
            handleChange={[Function]}
            id=""
            name="body"
            title="Descrição: "
            value=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "errors": Array [],
            "handleChange": [Function],
            "id": "",
            "name": "author",
            "placeholder": "",
            "title": "Autor: ",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "errors": Array [],
            "handleChange": [Function],
            "id": "",
            "name": "title",
            "placeholder": "",
            "title": "Titulo: ",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "errors": Array [],
            "handleChange": [Function],
            "id": "",
            "name": "category",
            "options": Array [
              Object {
                "label": "react",
                "value": "react",
              },
              Object {
                "label": "redux",
                "value": "redux",
              },
              Object {
                "label": "udacity",
                "value": "udacity",
              },
            ],
            "title": "Categoria: ",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "errors": Array [],
            "handleChange": [Function],
            "id": "",
            "name": "body",
            "title": "Descrição: ",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
